src/components/code-snippet/code-snippet.js

```diff
@@ -0,0 +1,68 @@
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentBySearch from '../../globals/js/mixins/init-component-by-search';
+import handles from '../../globals/js/mixins/handles';
+import on from '../../globals/js/misc/on';
+
+class CodeSnippet extends mixin(createComponent, initComponentBySearch, handles) {
+  /**
+   * CodeSnippet UI.
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @extends Handles
+   * @param {HTMLElement} element The element working as a CodeSnippet UI.
+   */
+
+  constructor(element, options) {
+    super(element, options);
+    this.manage(
+      on(this.element, 'click', event => {
+        this._handleClick(event);
+      })
+    );
+
+    this._initCodeSnippet();
+  }
+
+  _handleClick() {
+    this.element.parentElement.classList.toggle('bx--snippet--expand');
```

Comment:
Great to define `classExpanded` option (value is `bx--snippet--expand`) for more flexible custom theming, etc. Same for all classes/selectors.